File: Contracts/Model/Product/Response/OptionResponse.cs
namespace Contracts.Model.Product.Response;
public class OptionResponse
{
	public int Id { get; set; }
	public string Title { get; set; }
	public string? Description { get; set; }
	public double Price { get; set; }
	public string? Color { get; set; }
	public float? Size { get; set; }
	public float Weight { get; set; }
	public string? GenderFor { get; set; }
	public string? Material { get; set; }
	public string Condition { get; set; }
	public string? Images { get; set; } = string.Empty;
	public string Stock => StockCalculate().ToString();
	public int QuantityAvailable { get; set; }

	private StockStatusType StockCalculate()
	{
		if (QuantityAvailable == 0)
			return StockStatusType.OutOfStock;
		if (QuantityAvailable > 0 && QuantityAvailable < 10)
			return StockStatusType.LowStock;
		if (QuantityAvailable >= 10)
			return StockStatusType.InStock;
		return StockStatusType.InStock;
	}
}

File: ClientApp/src/app/interfaces/product/response/optionResponse.ts
import { MultipleImagesResponse } from "./multipleImagesResponse";

export interface OptionResponse {
	id: number;
	title: string;
	description: string | null;
	price: number;
	color: string | null;
	size: number | null;
	weight: number;
	genderFor: string | null;
	material: string | null;
	condition: string;
	images: MultipleImagesResponse[];
	stock: string;
	quantityAvailable: number;
}